### ISSUE TITLE
[Bug] Chunked Prefill + Radix Insert race corrupts KV pages (QSA, Qwen3.8-Flash-Next)

### ISSUE THREAD

**Issue by andreasknopke** [sgl-project/sglang#38319]
### Summary
A race condition between chunked-prefill KV-page insertion into the Radix Cache and pool page deallocation (retract/abort) produces silently corrupted KV pages. Once cached in the Radix tree, every subsequent request matching the same prefix immediately reads garbage attention data, causing an impossible-token loop (token ID `248319` = OOV/never-trained lm_head row) on the very first decode step.

### Symptoms
- First decode token after prefill is always the impossible token `248319` (tokenizer length = 248077, lm_head = 248320)
- `num_correct_drafts = 0`, `accept_len = 1` — the target model independently produces the bad token on its first forward
- The corrupted KV slot range is identical across repeated requests with the same prefix
- `flush_cache` resolves the issue temporarily (evicts the corrupt radix prefix)
- Re-appears at the same prefix once the corruption is re-triggered

### Hardware/Software
- **Hardware:** 1× DGX Spark (GB10, 128 GB, SM121)
- **Model:** `RadixArk/Qwen3.8-Flash-Next-NVFP4` (180B MoE, QSA sparse attention)
- **Serving:** SGLang with QSA KDA kernel + SM121 Triton fallback decode
- **KV cache dtype:** auto (NVFP4)
- **Speculative decoding:** EAGLE, 3 steps, topk=1, 4 draft tokens
- **Chunked prefill:** 2048 tokens/chunk
- **Page size:** 64 (compressed QSA full//ratio addressing)
- **Context length:** 262144, mem_fraction_static=0.79
- **Attention:** prefill=triton, decode=trtllm_mha
- **Mamba radix cache strategy:** extra_buffer

### Evidence

**Forensic dumps** available in our working directory. Key observations from 4 consecutive v6 (impossible-token) alarms:

```
DEGEN-IMPOSSIBLE-TOKEN rid=e80f09ea... bad_ids=[248319]
  req_pool_idx=3 seqlen=33854 kv_committed_len=33854
  prefix_len=33853 out_len=1
  accept_ids=[248319]
  slot_tail(128)=[160510, 160511, ..., 160637]  # identical range
  mamba_pool_idx=4 mamba_ping_pong=[15, 13]
```

Three subsequent requests with the **same** prefix_len (33853) and **identical** slot range `[160510...160637]` produced the **same** impossible token on their very first decode step. The only varying field is `mamba_pool_idx` — confirming the corruption lives in the **KV pages, not the Mamba state**.

The `slot_tail` shows a contiguous run of 128 KV slots all pointing to the same page range — exactly what you'd expect from a radix-cached prefix that was written by a corrupted chunked-prefill insert.

### Root Cause Analysis

The bug is in the interaction between **chunked prefill KV-page insertion** and the **Radix Cache**:

1. During chunked prefill, each chunk's KV pages are inserted into the QSA compressed-KV pool (`QSATokenToKVPool`) and simultaneously registered in the Radix tree
2. If the request is aborted or retracted mid-prefill, the `HybridReqToTokenPool` frees those pages
3. A race occurs when the Radix tree has already created a node referencing the newly-inserted KV pages, but the pages are freed by a concurrent pool deallocation before the radix insert is committed
4. The Radix node survives with dangling references — pointing to KV pages that were re-allocated to a different request or contain uninitialized data
5. Once the corrupt prefix is cached in the Radix, every request with the same history gets the same bad KV pages

This explains why the corruption:
- **Is stochastic** (depends on timing of chunk insert vs retract)
- **Persists across requests** (radix cache keeps the bad prefix)
- **Happens at specific context sizes** (33k, 145k — where pool fragmentation reaches specific states)
- **Wasn't present before the QSA KDA kernel patch** (the old trtllm/XQA routing masked the corruption by reading different slots)

### Why Token 248319?

Token 248319 is the **last row of lm_head.weight** (`[248320, 2560]`). It exists in the model's vocabulary but the tokenizer (`len=248077`) has no mapping for it. This row was never trained — it's padding from the vocabulary extension. When attention reads corrupted KV data, the logits for this position are noise, and it occasionally becomes the argmax. Once selected, the MTP draft sees the same corrupt KV context and reproduces it → "!!!!" loop.

### Proposed Fixes

1. **Atomic radix insert after pool commit** — The Chunked-Prefill insert should only register pages in the Radix tree after the pool confirms the write is complete and no `free` is pending for those pages
2. **Guard radix evict on impossible token** — When the v6 guard fires, evict the matched prefix from radix in addition to skipping insert (mitigation, not root fix)
3. **`--disable-chunked-prefix-cache` as test** — Disabling chunked radix inserts eliminates the race entirely (performance impact unknown for this model)

### Related
- Issue #36537 — Qwen3.8-Flash-Next thinking + tool parser loops (different root cause, same symptom class)
- Issue #36845 — KV corruption at high context length (related KV page management)
